src/pubkey.h

```diff
@@ -254,7 +254,11 @@ class XOnlyPubKey
     bool IsNull() const { return m_keydata.IsNull(); }
 
     /** Construct an x-only pubkey from exactly 32 bytes. */
-    explicit XOnlyPubKey(Span<const unsigned char> bytes);
+    constexpr explicit XOnlyPubKey(Span<const unsigned char> bytes)
+    {
+        assert(bytes.size() == 32);
```

Comment:
nit: while touching, might be worth updating to

```suggestion
        assert(bytes.size() == m_keydata.size());
```

Reply:
Taken in latest push.

Reply:
Further changed to just:
```C++
constexpr explicit XOnlyPubKey(Span<const unsigned char> bytes) : m_keydata{bytes} {}
```
Since `base_blob(Span<const unsigned char>)` does the exact length assert internally.